sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -53,6 +54,7 @@ case class ColumnarOverrideRules() extends ColumnarRule with Logging {
 object RapidsPluginUtils extends Logging {
   val CUDF_PROPS_FILENAME = "cudf-java-version-info.properties"
   val PLUGIN_PROPS_FILENAME = "rapids4spark-version-info.properties"
+  val DRIVER_TIMEZONE_KEY = "spark.rapids.driver.user.timezone"
```

Comment:
Nit: This key should be added to RapidsConf as an internal config with the corresponding lazy val accessor to document the usage in the namespace of the configs.  It can be made an optional config, and the executor code fetching it can throw if it fetches the conf key and gets `None`.  That eliminates the need for a new `get` method that throws.

Reply:
Thanks @jlowe for reviewing the PR. The reason I had kept it here was that I didn't want to expose this config to the user. If we added in RapidsConf as an internal config, then do we risk the config to be set by user which may not be the correct timezone. Please let me know if my understanding is wrong.

Reply:
Leaving the config completely undocumented does not remove the risk of what you're worried about.  The user could still end up setting that config, and if we fail to override it properly at the point where it matters then it will result in the same situation.

By documenting it as an internal config that will be programmatically set, we're educating the maintainers about this use of the config namespace.  I'm not worried about the user specifying this config, as the plugin should properly smash any previous value with the value on the driver side.  You can verify this by trying to set it yourself and see if it gets picked up or not.